executor/src/witgen/identity_processor.rs

```diff
@@ -162,15 +165,17 @@ impl<'a, 'b, T: FieldElement> IdentityProcessor<'a, 'b, T> {
             return result;
         }
 
-        for i in 0..self.machines.len() {
-            let (current, others) = self.machines.split(i);
-            if let Some(result) = current.process_plookup(
-                self.fixed_lookup,
-                identity.kind,
-                &left,
-                &identity.right,
-                others,
-            ) {
+        for i in 0..self.mutable_state.machines.len() {
```

Comment:
Would it make sense to change the iterator to directly return `(current, others)`?

Reply:
Or implement a function on `MutableState` that directly returns a `MutableState` with one machine taken out?

Reply:
Hmm, I just tried to implement this on `Machines`:
```rust
    pub fn iter_split<'c>(
        &'c mut self,
    ) -> impl Iterator<Item = (&'c mut KnownMachine<'a, T>, Machines<'a, 'c, T>)> {
        (0..self.len()).map(move |i| self.split(i))
    }
```

Gives me a lifetime error I'm not sure how to resolve:
```
error: captured variable cannot escape `FnMut` closure body
  --> executor/src/witgen/identity_processor.rs:50:38
   |
50 |         (0..self.len()).map(move |i| self.split(i))
   |                                    - ^^^^^^^^^^^^^ returns a reference to a captured variable which escapes the closure body
   |                                    |
   |                                    inferred to be a `FnMut` closure
   |
   = note: `FnMut` closures only have access to their captured variables while they are executing...
   = note: ...therefore, they cannot allow references to captured variables to escape
```

Reply:
If we find a solution, this can be a small refactoring PR :)